pyp2rpm/bin.py

```diff
@@ -159,8 +161,12 @@ def main(package, v, d, s, r, proxy, srpm, p, b, o, t, venv, autonc, sclize, **s
         register_console_log_handler()
 
     distro = o
-    if t in settings.KNOWN_DISTROS:
+    if t and os.path.splitext(t)[0] in settings.KNOWN_DISTROS:
         distro = t
+    elif t and not b or p:
```

Comment:
When I do `./mybin.py -p2 Sphinx` I get:
```
Usage: mybin.py [OPTIONS] PACKAGE

Error: Default python versions for template None are missing in settings, add them or use flags -b/-p to set python versions.
```

This line should probably be changed to `elif t and not (b or p):`